llvm/unittests/ADT/StringSetTest.cpp

```diff
@@ -73,4 +73,12 @@ TEST_F(StringSetTest, Contains) {
   EXPECT_FALSE(Set.contains("test"));
 }
 
+TEST_F(StringSetTest, Equal) {
+  StringSet<> A = {"A"};
+  StringSet<> B = {"B"};
+  ASSERT_TRUE(A != B);
+  ASSERT_FALSE(A == B);
+  ASSERT_TRUE(A == A);
```

Comment:
this seem redundant

Reply:
I added this check because there is a similar one in StringMap tests https://github.com/llvm/llvm-project/blob/main/llvm/unittests/ADT/StringMapTest.cpp#L451